Fix Eval so that it finds the classes directory for PR Snapshots

lloydmeta opened the pull request:
Closes #211 

Resolve the PR snapshot versions to just major minor in terms of target directory.

Tested locally using instructions in #211:

```
sbt:enumeratum-root> ++2.12.9-bin-f96c8d2-SNAPSHOT!
[info] Forcing Scala version to 2.12.9-bin-f96c8d2-SNAPSHOT on all projects.
[info] Reapplying settings...
[info] Set current project to enumeratum-root (in build file:/Users/lloyd/Documents/skala/enumeratum/)
sbt:enumeratum-root> coreJVMTests/test
[info] EnumJVMSpec:
[info] findValues Vector
[info] - should be in the same order that the objects were declared in
[info] ValueEnumJVMSpec:
[info] IntEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] LongEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] ShortEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] ByteEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] StringEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] CharEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] Run completed in 38 seconds, 896 milliseconds.
[info] Total number of tests run: 7
[info] Suites: completed 2, aborted 0
[info] Tests: succeeded 7, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 42 s, completed Jan 17, 2019 9:48:15 PM
```

cc @SethTisue

SethTisue commented on build.sbt:
```diff
@@ -456,7 +456,8 @@ lazy val resolverSettings = Seq(
     "Sonatype snapshots" at "https://oss.sonatype.org/content/repositories/snapshots",
-    "Sonatype releases" at "https://oss.sonatype.org/content/repositories/releases"
+    "Sonatype releases" at "https://oss.sonatype.org/content/repositories/releases",
+    "Scala PR" at "https://scala-ci.typesafe.com/artifactory/scala-pr-validation-snapshots"
```
it might be better not to merge this, since the more resolvers you have, the longer `update` takes. otherwise LGTM

lloydmeta replied:
True. Removed.

lloydmeta commented:
Updated this PR to use sbt-buildinfo to pass the directory along and removed a bunch of flakey regex-based logic. Still works

```scala
sbt:enumeratum-root> ++2.12.9-bin-f96c8d2-SNAPSHOT!
[info] Forcing Scala version to 2.12.9-bin-f96c8d2-SNAPSHOT on all projects.
[info] Reapplying settings...
[info] Set current project to enumeratum-root (in build file:/Users/lloyd/Documents/skala/enumeratum/)
sbt:enumeratum-root> coreJVMTests/test
[info] Updating macrosJVM...
[info] Updating coreJVM...
[warn] Resolving a snapshot version. It's going to be slow unless you use `updateOptions := updateOptions.value.withLatestSnapshots(false)` options.
[info] Out of 1 candidates we found for org.scala-lang#scala-library;2.12.9-bin-f96c8d2-SNAPSHOT in Scala PR, we are choosing Scala PR.
[warn] Resolving a snapshot version. It's going to be slow unless you use `updateOptions := updateOptions.value.withLatestSnapshots(false)` options.
[info] Out of 1 candidates we found for org.scala-lang#scala-reflect;2.12.9-bin-f96c8d2-SNAPSHOT in Scala PR, we are choosing Scala PR.
[warn] Resolving a snapshot version. It's going to be slow unless you use `updateOptions := updateOptions.value.withLatestSnapshots(false)` options.
[info] Out of 1 candidates we found for org.scala-lang#scala-compiler;2.12.9-bin-f96c8d2-SNAPSHOT in Scala PR, we are choosing Scala PR.
[info] Done updating.
[warn] Resolving a snapshot version. It's going to be slow unless you use `updateOptions := updateOptions.value.withLatestSnapshots(false)` options.
[info] Out of 1 candidates we found for org.scala-lang#scala-library;2.12.9-bin-f96c8d2-SNAPSHOT in Scala PR, we are choosing Scala PR.
[warn] Resolving a snapshot version. It's going to be slow unless you use `updateOptions := updateOptions.value.withLatestSnapshots(false)` options.
[info] Out of 1 candidates we found for org.scala-lang#scala-reflect;2.12.9-bin-f96c8d2-SNAPSHOT in Scala PR, we are choosing Scala PR.
[warn] Resolving a snapshot version. It's going to be slow unless you use `updateOptions := updateOptions.value.withLatestSnapshots(false)` options.
[info] Out of 1 candidates we found for org.scala-lang#scala-compiler;2.12.9-bin-f96c8d2-SNAPSHOT in Scala PR, we are choosing Scala PR.
[info] Done updating.
[info] Updating coreJVMTests...
[warn] Resolving a snapshot version. It's going to be slow unless you use `updateOptions := updateOptions.value.withLatestSnapshots(false)` options.
[info] Out of 1 candidates we found for org.scala-lang#scala-library;2.12.9-bin-f96c8d2-SNAPSHOT in Scala PR, we are choosing Scala PR.
[warn] Resolving a snapshot version. It's going to be slow unless you use `updateOptions := updateOptions.value.withLatestSnapshots(false)` options.
[info] Out of 1 candidates we found for org.scala-lang#scala-reflect;2.12.9-bin-f96c8d2-SNAPSHOT in Scala PR, we are choosing Scala PR.
[warn] Resolving a snapshot version. It's going to be slow unless you use `updateOptions := updateOptions.value.withLatestSnapshots(false)` options.
[info] Out of 1 candidates we found for org.scala-lang#scala-compiler;2.12.9-bin-f96c8d2-SNAPSHOT in Scala PR, we are choosing Scala PR.
[info] Done updating.
[warn] There may be incompatibilities among your library dependencies.
[warn] Run 'evicted' to see detailed eviction warnings
[info] Compiling 4 Scala sources to /Users/lloyd/Documents/skala/enumeratum/enumeratum-core/.jvm/target/scala-2.12/classes ...
[info] Compiling 3 Scala sources to /Users/lloyd/Documents/skala/enumeratum/macros/.jvm/target/scala-2.12/classes ...
[info] Done compiling.
[info] Done compiling.
[info] Compiling 1 Scala source to /Users/lloyd/Documents/skala/enumeratum/enumeratum-core-jvm-tests/target/scala-2.12/classes ...
[info] Done compiling.
[info] Compiling 3 Scala sources to /Users/lloyd/Documents/skala/enumeratum/enumeratum-core-jvm-tests/target/scala-2.12/test-classes ...
[info] Done compiling.
[info] EnumJVMSpec:
[info] findValues Vector
[info] - should be in the same order that the objects were declared in
[info] ValueEnumJVMSpec:
[info] IntEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] LongEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] ShortEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] ByteEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] StringEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] CharEnum withValue
[info] - should return proper members for valid values but throw otherwise
[info] Run completed in 50 seconds, 688 milliseconds.
[info] Total number of tests run: 7
[info] Suites: completed 2, aborted 0
[info] Tests: succeeded 7, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 149 s, completed Jan 18, 2019 11:02:42 AM
```
